### ISSUE TITLE
Peforming mash with eQTL results from different variants & gene sets across conditions

### ISSUE THREAD

**Issue by kimbioinfo** [stephenslab/mashr#122]
Hello, I'm trying to use mashr to robustly compare my own eQTL results from three conditions.
Actually, I've already got three eQTL result sets from the condition-by-condition alaysis.
Those eQTL results were calculated from different variant and gene sets.
I applied GTEx gene filtering criteria and general genotype filtering steps on each condition, and I didn't do intersection for them.

For example, I used datasets like below:
Condition A - 26,000 genes / 6,500,000 SNPs
Condition B - 24,000 genes / 6,000,000 SNPs
Condition C - 23,000 genes / 6,300,000 SNPs
(fyi, Intersection - 21,000 genes / 5,300,000 SNPs)

Thus, I extracted overlapping SNP-gene pairs from the three eQTL results and put effect sizes and standard errors of those pairs into mashr.
In my guess, there would be trace changes in effect sizes and standard errors followed by somewhat changes in covariance made by mashr.
In this case, will there be any significiant chages in posterior mean and standard dev. values?
Should I use same variant and gene sets across all conditions for eQTL analysis?

Thanks for providing a good tool and managing.

**Comment by pcarbo** [comment]
> Should I use same variant and gene sets across all conditions for eQTL analysis?

Yes, the effect estimates should come from the same gene-SNP pairs measured in different conditions.

> In my guess, there would be trace changes in gene expression values due to normalization steps, followed by trace changes in effect sizes and standard errors.

> If there are problems in the original analysis that might produce false positives, I would not expect mashr to correct these issues; you should first address these problems before running mashr.
